src/classes/Contacts.cls

```diff
@@ -0,0 +1,748 @@
+                }
+            } else {
+                householdInserts.add(con);
+            }
+        }
+        return householdInserts;
+    }
+
+    private Boolean hasAccount(Contact con) {
+        return con.AccountId != null;
+    }
+
+    private Boolean isIndividualAccount(Account acc) {
+        return acc.npe01__SYSTEMIsIndividual__c == true;
+    }
+
+    public TDTM_Runnable.DmlWrapper handleAfterUpdate(TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        //if its an update, we need to get some data about their
+        //old Household for processing
+        buildOldHouseholdCollections();
+        processHouseholdAndNamingFieldChangesAfterUpdate();
+
+        dmlWrapper.objectsToDelete.addAll(emptyHouseholdsToDelete);
+
+        contactsNeedingHouseholds.addAll(getContactsNeedingHouseholds(contactEvaluations));
+        insertHouseholdsForContacts(contactsNeedingHouseholds, false);
+
+        // Note: This set will contain Household__c record Id's only (i.e., legacy models)
+        if (householdIdsNeedingRollup.size() > 0) {
+            calculateRollups();
+        }
+
+        if (householdsNeedingNamesUpdated.size() > 0) {
+            updateHouseholdNames(householdsNeedingNamesUpdated);
+        }
+        return dmlWrapper;
+    }
+
+    public void rollupAndRenameHouseholdsAfterDelete(TDTM_Runnable.DmlWrapper dmlWrapper) {
+        buildDeletedAndMergedContactCollections();
+
+        if (mergedWinningContact.size() > 0) {
+            // this fires off opp rollups, which we aren't going to move to dmlWrapper!
+            contactMergeUpdate(mergedWinningContact);
+        }
+
+        if (deletedContactsInHouseholds.size() > 0) {
+            deleteOrRenameHouseholdsWhenContactDeleted(
+                    deletedContactsInHouseholds, dmlWrapper);
+        }
+    }
+
+    private void processHouseholdAndNamingFieldChangesAfterUpdate() {
+        for (Contact con : (List<Contact>) getRecords()) {
+            //if we're doing an update, pile up all of the households into a list
+            //and send it to our naming code for action
+            //first contact's names to their old names, and see if anything changed
+            //salutation
+            if (!hasHousehold(con)) {
+                handleContactHasNoHousehold(con);
+            }
+
+            //if they've changed households, we need to query the old
+            //household to see if anybody is left, if so, send it for renaming
+            //if not, delete it.
+            if (householdChanged(con)) {
+                handleHouseholdChanged(con);
+            } else if (isHouseholdNamingFieldsChanged(con)) {
+                householdsNeedingNamesUpdated.add(householdIdFor(con));
+            }
+        }
+    }
+
+    private Boolean isHouseholdNamingFieldsChanged(Contact con) {
+        return isNamingExclusionsChanged(con) || isNamingFieldChanged(con);
+    }
+
+    Set<String> householdNamingFieldsInUse;
```

Comment:
Let's move the class variable to the top of the class.

Reply:
👍🏽